dcpy/utils/postgres.py

```diff
@@ -8,41 +8,142 @@
 
 
+class PostgresClient:
+    def __init__(
+        self,
+        schema: Optional[str] = None,
+        database: str = os.environ["BUILD_ENGINE_DB"],
+        server_url: str = os.environ["BUILD_ENGINE_SERVER"],
+    ):
+        self.server_url = server_url
+        self.database = database
+        self.schema = _format_schema_name(schema)
+        self.engine_uri = generate_engine_uri(
+            server_url=server_url, database=database, schema=schema
+        )
+        self.engine = create_engine(
+            self.engine_uri,
+            isolation_level="AUTOCOMMIT",
+        )
+        self.create_schema(self.schema)
+
+    def execute_query(self, query: str, placeholders: Optional[dict] = None) -> None:
+        if placeholders is None:
+            placeholders = {}
+        with self.engine.connect() as connection:
+            connection.execute(statement=text(query), parameters=placeholders)
+
+    def execute_select_query(
+        self,
+        query: str,
+        placeholders: Optional[dict] = None,
+    ) -> pd.DataFrame:
+        if placeholders is None:
+            placeholders = {}
+        with self.engine.connect() as sql_conn:
+            select_records = pd.read_sql(
+                sql=text(query), con=sql_conn, params=placeholders
+            )
+        return select_records
+
+    def create_postigs_extension(self) -> None:
+        self.execute_query("CREATE EXTENSION POSTGIS")
+
+    def vacuum_database(self) -> None:
+        self.execute_query("VACUUM (ANALYZE)")
+
+    def drop_schema(self, schema_name: str) -> None:
+        self.execute_query(
+            "DROP SCHEMA IF EXISTS :schema_name CASCADE",
+            {"schema_name": AsIs(schema_name)},
+        )
+
+    def create_schema(self, schema_name: str) -> None:
+        self.execute_query(
+            "CREATE SCHEMA IF NOT EXISTS :schema_name",
+            {"schema_name": AsIs(schema_name)},
+        )
+
+    def get_schemas(self) -> list:
+        schema_names = self.execute_select_query(
+            """
+            SELECT schema_name FROM information_schema.schemata
+            """
+        )
+        return sorted(schema_names["schema_name"].to_list())
+
+    def get_schema_tables(self, table_schema: str) -> list:
+        select_table_names = self.execute_select_query(
+            """
+            SELECT table_name FROM information_schema.tables WHERE table_schema = :table_schema
+            """,
+            {"table_schema": table_schema},
+        )
+        all_table_names = sorted(select_table_names["table_name"].to_list())
+        postgis_tables = [
+            "spatial_ref_sys",
+            "geography_columns",
+            "geometry_columns",
+        ]
+        table_names = [
+            table_name
+            for table_name in all_table_names
+            if table_name not in postgis_tables
+        ]
+        return table_names
+
+    def get_table_columns(self, table_schema: str, table_name: str) -> list:
```

Comment:
table_schema should maybe have default of the schema of the client? This would simplify calls in qa app to just `client.get_table_columns(f"{table_name}")`

Applies to other functions as well

Reply:
I'm thinking maybe no argument and no default since (for now) the client is designed to be limited to a single schema and `public`?

Reply:
I think there are definitely possible cases in the future where we'd like to query across schemas, in general I would prefer the client is scoped to a specific database but there's nothing limiting, say, referencing other schemas explicitly in a query, so we don't necessarily need to impose that limitation on some of these functions.